cql3/Cql.g

```diff
@@ -219,37 +218,39 @@ struct uninitialized {
         return token->getText();
     }
 
-    std::map<sstring, sstring> convert_property_map(shared_ptr<cql3::maps::literal> map) {
-        if (!map || map->entries.empty()) {
+    std::map<sstring, sstring> convert_property_map(const cql3::expr::expression& map_expr) {
+        auto map = std::get<cql3::expr::collection_constructor>(map_expr);
+        if (map.elements.empty()) {
```

Comment:
I agree, but I wanted to avoid all unnecessary changes and make this as close as possible to a mechanical translation (of course I did make much larger changes when choosing to merge some types, for example maps::literal and sets::literal becoming collection_constructor). I'm happy to follow up with this simplification.